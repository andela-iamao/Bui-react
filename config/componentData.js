module.exports = [{"name":"Button","description":"Button Component","props":{"children":{"type":{"name":"node"},"required":true,"description":"Child elements for the button component"},"onClick":{"type":{"name":"func"},"required":false,"description":""},"type":{"type":{"name":"enum","value":[{"value":"'primary'","computed":false},{"value":"'secondary'","computed":false},{"value":"'danger'","computed":false}]},"required":false,"description":""}},"code":"import React from 'react';\nimport PropTypes from 'prop-types';\n\n\nconst colorScheme = {\n    primary: '#236167',\n    secondary: '#000000',\n    danger: '#AA3939'\n}\n\n/**\n * Button Component\n */\nfunction Button({ children, ...props}) {\n    return (\n        <button {...props} style={style(props.type)}>\n            {children}\n        </button>\n    );\n}\n\nconst style = (type) => ({\n    color: '#FFFFFF',\n    backgroundColor: colorScheme[type],\n    padding: 10,\n    width: 120,\n    border: 'none'\n});\n\nButton.propTypes = {\n    /**\n     * Child elements for the button component\n     */\n    children: PropTypes.node.isRequired,\n    onClick: PropTypes.func,\n    type: PropTypes.oneOf(['primary', 'secondary', 'danger'])\n}\n\nexport default Button;\n","examples":[{"name":"DangerButton","description":"Danger Button component","code":"import React from 'react';\nimport Button from 'bui-react/Button';\n\n/**\n * Danger Button component\n */\nexport default () => {\n    return (\n        <Button type=\"danger\">Danger Button!</Button>\n    );\n}\n"},{"name":"PrimaryButton","description":"Primary Button component","code":"import React from 'react';\nimport Button from 'bui-react/Button';\n\n/**\n * Primary Button component\n */\nexport default () => {\n    return (\n        <Button type=\"primary\">Primary Button!</Button>\n    );\n}\n"},{"name":"SecondaryButton","description":"Secondary Button component","code":"import React from 'react';\nimport Button from 'bui-react/Button';\n\n/**\n * Secondary Button component\n */\nexport default () => {\n    return (\n        <Button type=\"secondary\">Secondary Button!</Button>\n    );\n}\n"}]}]