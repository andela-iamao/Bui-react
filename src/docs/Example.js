import React from 'react';
import PropTypes from 'prop-types';

import { autobind } from 'core-decorators';

import Snippet from './Snippet';


class Example extends React.Component {
    constructor(props) {
        super(props);
        this.state = { showCode: false };
    }

    @autobind toggleCode(event) {
        event.preventDefault();
        this.setState((prevState) => ({
            showCode: !prevState.showCode
        }));
    }

    render() {
        const { showCode } = this.state;
        const { code, description, name } = this.props.example;

        const ExampleComponent = require(`./examples/${this.props.componentName}/${name}`).default;
        return (
            <div>
                {description && <h4>{description}</h4>}

                <ExampleComponent />

                <p>
                    <a href="#" onClick={this.toggleCode}>
                        {showCode ? "Hide": "show"} Code
                    </a>
                </p>


                {showCode && <Snippet>{code}</Snippet>}
            </div>
        )
    }
}

Example.propTypes = {
    example: PropTypes.object.isRequired,
    componentName: PropTypes.string.isRequired
}


export default Example;
