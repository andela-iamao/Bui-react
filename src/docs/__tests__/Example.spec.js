import React from 'react';
import renderer from 'react-test-renderer';

import { mount } from 'enzyme';

import Example from '../Example';
import componentData from '../../../config/jest/componentDataMock';

const DATA = componentData[0];


describe('Docs:Example', () => {

    describe('Snapshot Test', () => {
        test('renders component without description', () => {
            const data = { ...componentData[0]};
            data.description = null;
            const tree = renderer.create(
                <Example
                    example={data.examples[0]}
                    componentName={data.name}
                />
            ).toJSON();
            expect(tree).toMatchSnapshot();
        });

        test('renders component with description', () => {
            const data = { ...componentData[0]};
            const tree = renderer.create(
                <Example
                    example={data.examples[0]}
                    componentName={data.name}
                />
            ).toJSON();
            expect(tree).toMatchSnapshot();
        });

    });

    describe('Integration Test', () => {
        test('renders and mount the component its child components', () => {
            const wrapper = mount(
                <Example
                    example={DATA.examples[0]}
                    componentName={DATA.name}
                />
            )
            expect(wrapper.find({ id: 'bui-react-docs-component-example-description' })).toHaveLength(1);
            expect(wrapper.find({ id: 'bui-react-docs-component-example-code' })).toHaveLength(1);
        });

        test('renders the example code on toggle', () => {
            const wrapper = mount(
                <Example
                    example={DATA.examples[0]}
                    componentName={DATA.name}
                />
            )
            wrapper.find({ id: 'bui-react-docs-component-example-code-toggle-btn' }).simulate('click');
            expect(wrapper.find({ id: 'bui-react-docs-component-code-snippet-container' })).toHaveLength(1);
        });

    });

});

