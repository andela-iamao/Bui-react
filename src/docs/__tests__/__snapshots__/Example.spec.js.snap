// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Docs:Example Snapshot Test renders component with description 1`] = `
<div
  id="bui-react-docs-component-example-container"
>
  <h4
    id="bui-react-docs-component-example-description"
  >
    Default Button component
  </h4>
  <button
    data-radium={true}
    style={
      Object {
        "backgroundColor": "#FFF",
        "border": "1px solid #333",
        "color": "#333",
        "minWidth": 150,
        "padding": 10,
      }
    }
  >
    Hello!
  </button>
  <p
    id="bui-react-docs-component-example-code"
  >
    <span
      className="link"
      id="bui-react-docs-component-example-code-toggle-btn"
      onClick={[Function]}
    >
      show
       Code
    </span>
  </p>
  <hr />
</div>
`;

exports[`Docs:Example Snapshot Test renders component without description 1`] = `
<div
  id="bui-react-docs-component-example-container"
>
  <h4
    id="bui-react-docs-component-example-description"
  >
    Default Button component
  </h4>
  <button
    data-radium={true}
    style={
      Object {
        "backgroundColor": "#FFF",
        "border": "1px solid #333",
        "color": "#333",
        "minWidth": 150,
        "padding": 10,
      }
    }
  >
    Hello!
  </button>
  <p
    id="bui-react-docs-component-example-code"
  >
    <span
      className="link"
      id="bui-react-docs-component-example-code-toggle-btn"
      onClick={[Function]}
    >
      show
       Code
    </span>
  </p>
  <hr />
</div>
`;
